src/suggestionlistmodel.cpp

```diff
@@ -46,6 +46,11 @@ QVariant SuggestionListModel::data(const QModelIndex &index, int role) const
                 return library->getZimIcon(zimId, defaultIcon);
             }
             break;
+        case Qt::SizeHintRole:
+            /* See resources/css/popup.css QHeaderView::section.
+               Take line-height + 5 padding top and bottom.
+            */
+            return QSize(0, 34);
```

Comment:
If this value cannot be computed from values obtained from CSS, then you can at least minimize the maintenance burden by defining a few constants in a dedicated source file and cross referencing it with the css file so that at most 2 locations have to be updated if/when any of those values changes. For example

**css_constants.h**
```c++ 

namespace CSS
{

// Need a good explanation of why this file is needed and the logical connection
// of these values to their counterparts in the CSS files must be emphasized.
// A cross reference to each constant from this file must be put beside the respective
// property in the CSS file.

// QHeaderView::section:line-height
const int SUGGESTION_LINE_HEIGHT = 24;

// QHeaderView::section:padding-top
const int SUGGESTION_LINE_PADDING_TOP = 5;
}
```

and then

```c++
return QSize(0, CSS::SUGGESTION_LINE_HEIGHT + CSS::SUGGESTION_LINE_PADDING_TOP + CSS::SUGGESTION_LINE_PADDING_BOTTOM);
```

Note that the current value of 34 doesn't match the formula from the code comment and the actual CSS values (which rather evaluates to 39).

All other similar magic numbers introduced in C++ code by this PR must be eliminated in a similar way.

Reply:
@veloman-yunkan I furthur propose we can use multi-level namespaces to further structure the CSS. If a file wants to use, for example, CSS::QTabBar::QWidget, they can do an aliasing to get a shorter name. 

One question, why does it evaluate it to 39?